python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+    parser.add_argument("-o", "--outputs", help="path to the .npz output file")
+    parser.add_argument(
+        "--print-time", action="store_true", help="record and print the execution time(s)"
+    )
+    parser.add_argument(
+        "--print-top",
+        metavar="N",
+        type=int,
+        help="print the top n values and indices of the output tensor",
+    )
+    parser.add_argument(
+        "--profile", action="store_true", help="generate profiling data from the runtime execution"
+    )
+    parser.add_argument("--repeat", metavar="N", type=int, default=1, help="repeat the run n times")
+    parser.add_argument(
+        "--rpc-key",
+        nargs=1,
+        help="the RPC tracker key of the target device",
+    )
+    parser.add_argument(
+        "--rpc-tracker",
+        nargs=1,
+        help="hostname (required) and port (optional, defaults to 9090) of the RPC tracker, "
+        "e.g. '192.168.0.100:9999'",
+    )
+    parser.add_argument("FILE", help="path to the compiled module file")
+
+
+def drive_run(args):
+    """Invoke runner module with command line arguments
+
+    Parameters
+    ----------
+    args: argparse.Namespace
+        Arguments from command line parser.
+    """
+    inputs = {}
+    if args.inputs:
+        inputs = np.load(args.inputs)
+
+    rpc_hostname, rpc_port = common.tracker_host_port_from_cli(args.rpc_tracker)
+
+    outputs, times = run_module(
+        args.FILE,
+        rpc_hostname,
+        rpc_port,
+        args.rpc_key,
+        inputs=inputs,
+        device=args.device,
+        fill_mode=args.fill_mode,
+        repeat=args.repeat,
+        profile=args.profile,
+    )
+
+    if args.print_time:
+        stat_table = format_times(times)
+        # print here is intentional
+        print(stat_table)
+
+    if args.print_top:
+        top_results = get_top_results(outputs, args.print_top)
+        # print here is intentional
+        print(top_results)
+
+    if args.outputs:
+        # Save the outputs
+        np.savez(args.outputs, **outputs)
+
+
+def get_input_info(graph_str, params):
+    """Return the 'shape' and 'dtype' dictionaries for the input
+    tensors of a compiled module.
+
+    Parameters
+    ----------
+    graph_str : str
+        JSON graph of the module serialized as a string.
+    params : bytearray
+        Params serialized as a bytearray.
+
+    Returns
+    -------
+    shape_dict : dict
+        Shape dictionary - {input_name: tuple}.
+    dtype_dict : dict
+        dtype dictionary - {input_name: dtype}.
+    """
+    # NOTE - We can't simply get the input tensors from a TVM graph
+    # because weight tensors are treated equivalently. Therefore, to
+    # find the input tensors we look at the 'arg_nodes' in the graph
+    # (which are either weights or inputs) and check which ones don't
+    # appear in the params (where the weights are stored). These nodes
+    # are therefore inferred to be input tensors.
```

Comment:
- Put this note to the docstring, like https://github.com/apache/incubator-tvm/blob/master/python/tvm/relay/op/transform.py#L414
- cc @zhiics @masahi to help check this logic.

Reply:
This logic is okay to me for now. In the long run, we may want to change graph runtime a bit and let it save inputs and params separately. params can be obtained from MetadataModule directly.

Reply:
Very interesting! I didn't know `.. note::` before. Done now.